### PR TITLE
Refactor: fully refactored original obsolete lapack hsolver

### PR DESCRIPTION
**Pull request by Critsium-xy**
This is the first step refactoring hsolver_lcao module. The serial lcao caculator has been recovered by refactoring the lapack hsolver.
No unit test and other things like document modification included. This will be submitted in another pull request.

image

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -116,6 +116,8 @@ elseif(NOT ENABLE_LCAO AND ENABLE_MPI)
   set(ABACUS_BIN_NAME abacus_pw)
 elseif(NOT ENABLE_LCAO AND NOT ENABLE_MPI)
   set(ABACUS_BIN_NAME abacus_pw_serial)
+elseif(ENABLE_LCAO AND NOT ENABLE_MPI)
+  set(ABACUS_BIN_NAME abacus_serial)
```
Comment (mohanchen) [inline CMakeLists.txt:120]: Is it better to change 'abacus_serial' to 'abacus_lcao_serial'?

File: source/module_basis/module_ao/ORB_control.cpp [hunk 2]
```diff
@@ -32,7 +32,9 @@ ORB_control::ORB_control() : setup_2d(false)
 }
 ORB_control::~ORB_control()
 {
+#ifdef __MPI
```
Comment (mohanchen) [inline source/module_basis/module_ao/ORB_control.cpp:35]: Could you put "#ifdef__MPI and #endif" into the function of Cblacs_exit(1)？

File: source/module_basis/module_ao/ORB_control.cpp [hunk 3]
```diff
@@ -111,7 +113,7 @@ void ORB_control::setup_2d_division(std::ofstream& ofs_running, std::ofstream& o
         // is determined in 'divide_HS_2d' subroutine
         this->divide_HS_2d(DIAG_WORLD, ofs_running, ofs_warning);
 #else
-        ModuleBase::WARNING_QUIT("LCAO_Matrix::init", "diago method is not ready.");
+        this->divide_HS_2d(ofs_running, ofs_warning);
```
Comment (mohanchen) [inline source/module_basis/module_ao/ORB_control.cpp]: you can add a note in front of the 116 line, such as "zhangxiaoyang enable the serial version of LCAO， 2024-07-06"

File: source/module_esolver/esolver.cpp [hunk 4]
```diff
@@ -179,7 +179,11 @@ void init_esolver(ESolver*& p_esolver)
 
 void clean_esolver(ESolver*& pesolver)
 {
+// No release step for serial version
+// It will be modified later
```
Comment (mohanchen) [inline source/module_esolver/esolver.cpp]: add your name here,

File: source/module_basis/module_ao/ORB_control.cpp (hunk 2 above)
Comment (Critsium-xy) [reply source/module_basis/module_ao/ORB_control.cpp:35]: Cblacs_exit is an external function and may be difficult to modify

**Review by mohanchen — APPROVED** [review]

**Pull request merged**